torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -331,6 +331,12 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
   requestMessage.setId(nextMessageID_++);
   pendingResponseMessage[requestMessage.id()] = futureResponseMessage;
 
+  futureResponseMessage->futMsg.addCallback([this]() {
```

Comment:
I saw there is another `addCallback` in `TensorPipeAgent::respond`, what about that one?

Reply:
We could add a similar check there too, but I think it's less relevant: the future in the `respond` method is created and marked as complete by the RequestCallback, and I think we can trust that it is done correctly. The future we're dealing with here is marked as complete by the agent itself, and it's against errors in the agent that I'm adding this check.